src/layer/__tests__/retries.ts

```diff
@@ -0,0 +1,137 @@
+import 'jest-setup';
+import axios from 'axios';
+import MockAdapter from 'axios-mock-adapter';
+
+import {
+  BBox,
+  CRS_EPSG4326,
+  S1GRDAWSEULayer,
+  AcquisitionMode,
+  Polarization,
+  Resolution,
+  OrbitDirection,
+  LinkType,
+} from 'src';
+
+const mockNetwork = new MockAdapter(axios);
+
+test('S1GRDLayer.findTiles retries correctly on network errors', async () => {
+  // const [ hasMoreFixture, sensingTimeFixture, expectedSensingTimeFixture ] = [false, '2018-11-28T11:12:13Z', new Date(Date.UTC(2018, 11 - 1, 28, 11, 12, 13))];
+  const hasMoreFixture = false;
+  const sensingTimeFixture = '2018-11-28T11:12:13Z';
+  const expectedSensingTimeFixture = new Date(Date.UTC(2018, 11 - 1, 28, 11, 12, 13));
+
+  const fromTime = new Date(Date.UTC(2018, 11 - 1, 22, 0, 0, 0));
+  const toTime = new Date(Date.UTC(2018, 12 - 1, 22, 23, 59, 59));
+  const bbox = new BBox(CRS_EPSG4326, 19, 20, 20, 21);
+  const layer = new S1GRDAWSEULayer({
+    instanceId: 'INSTANCE_ID',
+    layerId: 'LAYER_ID',
+    acquisitionMode: AcquisitionMode.IW,
+    polarization: Polarization.DV,
+    resolution: Resolution.HIGH,
+  });
+
+  // we need to adjust jest timeout until we have support for setting the delay when retrying, or
+  // the test will time out:
+  jest.setTimeout(7000);
+
+  mockNetwork.onPost().replyOnce(500);
```

Comment:
Shouldn't this be "binded"  to endpoint used by `findTiles`? I know it doesn't matter in this case, but this code will probably be used as a template for other tests. fftr

Reply:
On one hand it would make sense to make sure the network calls are completely correct, but on the other hand the tests should have well-defined scope. In this case the scope is just retrying of network requests - we shouldn't know about the specifics of the call being made. If we are testing URL we should probably also test request parameters...

However, we _are_ using a specific network response and comparing it to known return values (to make sure that the method really succeeded). So I'm not 100% sure what the best approach would be here.

I would suggest we leave this as-is, and refactor later if needed, does that sound OK to you?